base/compiler/ssair/inlining.jl

```diff
@@ -1351,6 +1348,8 @@ function handle_cases!(ir::IRCode, idx::Int, stmt::Expr, @nospecialize(atype),
     if fully_covered && length(cases) == 1
         handle_single_case!(ir, idx, stmt, cases[1].item, todo, params)
     elseif length(cases) > 0
+        isa(atype, DataType) || return nothing
+        all(case::InliningCase->isa(case.sig, DataType), cases) || return nothing
```

Comment:
It is also okay if fully_covers, right?

Reply:
I think you already found it, but this condition is required at https://github.com/JuliaLang/julia/blob/0a3d86cf423f46b0948fd14cedc8a69cc7da7d5a/base/compiler/ssair/inlining.jl#L477

Reply:
ah, that seems slightly too aggressive at asserting the type, but probably does not ever matter. We would need to change that to use `fieldtype(metharg, n)` instead